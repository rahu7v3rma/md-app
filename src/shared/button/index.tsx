import React, { FunctionComponent } from 'react';
import {
    GestureResponderEvent,
    StyleProp,
    StyleSheet,
    TouchableOpacity,
    ViewStyle
} from 'react-native';

import { Colors } from '@/theme/colors';
import { COMMON } from '@/utils/common';

type Props = {
    children?: React.ReactNode;
    primary?: boolean;
    bordered?: boolean;
    rounded?: boolean;
    block?: boolean;
    style?: StyleProp<ViewStyle>;
    disabled?: boolean;
    onPress?: (e: GestureResponderEvent) => void;
    testID?: string;
};

const Button: FunctionComponent<Props> = ({
    children,
    primary = false,
    bordered = true,
    rounded,
    block,
    style,
    disabled,
    onPress,
    testID
}: Props) => {
    return (
        <TouchableOpacity
            testID={testID}
            style={[
                buttonStyle.wrapper,
                bordered && buttonStyle.bordered,
                primary && buttonStyle.primary,
                disabled && buttonStyle.disabled,
                rounded && buttonStyle.rounded,
                block && buttonStyle.block,
                style
            ]}
            disabled={disabled}
            activeOpacity={disabled ? 1 : 0.75}
            onPress={onPress}
        >
            {children}
        </TouchableOpacity>
    );
};

export default Button;

const buttonStyle = StyleSheet.create({
    wrapper: {
        justifyContent: 'center',
        alignItems: 'center',
        paddingHorizontal: 15.5,
        paddingVertical: COMMON.isIos ? 24 : 17.5,
        borderRadius: 20,
        margin: 0
    },
    bordered: {
        borderWidth: 1,
        borderBottomWidth: 5,
        borderColor: Colors.theme.primary
    },
    primary: {
        backgroundColor: Colors.theme.primary,
        borderColor: Colors.theme.primary
    },
    disabled: {
        backgroundColor: Colors.button.app_button_disabled_bg,
        borderColor: Colors.button.app_button_disabled_border
    },
    rounded: {
        borderRadius: 100,
        backgroundColor: Colors.button.app_btn_light_color
    },
    block: {
        width: '100%'
    }
});
