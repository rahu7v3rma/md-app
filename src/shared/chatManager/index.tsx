import React, { FunctionComponent, useState } from 'react';

import { AppChatProvider } from '@/contexts/appChat';

import { useChatNotifications, useInitChatClient } from './hooks';

type Props = {
    children?: React.ReactNode;
};

const ChatManager: FunctionComponent<Props> = ({ children }: Props) => {
    const { chatClient } = useInitChatClient();

    const [activeChatChannel, setActiveChatChannel] = useState<string | null>(
        null
    );

    useChatNotifications(chatClient, activeChatChannel);

    return (
        <AppChatProvider value={{ chatClient, setActiveChatChannel }}>
            {children}
        </AppChatProvider>
    );
};

export default ChatManager;
