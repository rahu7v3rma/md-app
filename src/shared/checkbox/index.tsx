import React, { FunctionComponent } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';

import { CheckIcon } from '@/assets/svgs';
import { Colors } from '@/theme/colors';

type Props = {
    checked: boolean;
    onChange: (value: boolean) => void;
};

const Checkbox: FunctionComponent<Props> = ({ checked, onChange }: Props) => {
    return (
        <TouchableOpacity
            style={[checkboxStyle.wrapper, checked && checkboxStyle.checked]}
            activeOpacity={0.8}
            onPress={() => onChange(!checked)}
        >
            {checked && (
                <CheckIcon color={Colors.extras.white} width={18} height={18} />
            )}
        </TouchableOpacity>
    );
};

export default Checkbox;

const checkboxStyle = StyleSheet.create({
    wrapper: {
        backgroundColor: Colors.text.white,
        borderWidth: 1.5,
        borderColor: Colors.button.app_button_disabled_border,
        width: 28,
        height: 28,
        borderRadius: 9,
        alignItems: 'center',
        justifyContent: 'center'
    },
    checked: {
        backgroundColor: Colors.text.green,
        borderWidth: 0
    }
});
