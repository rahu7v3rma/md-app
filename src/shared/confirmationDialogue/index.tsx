import React, { FunctionComponent, RefObject } from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import RBSheet from 'react-native-raw-bottom-sheet';

import Button from '@/shared/button';
import Text, { Size } from '@/shared/text';
import { Colors } from '@/theme/colors';

type Props = {
    bottomSheetRef: RefObject<RBSheet>;
    title: string;
    subTitle?: string;
    dismissBtnTitle?: string;
    confirmBtnTitle?: string;
    onDismissBtnHandler?: () => void;
    onConfirmBtnHandler?: () => void;
    confirmBtnStyles?: StyleProp<ViewStyle>;
};

const ConfirmationDialogue: FunctionComponent<Props> = ({
    bottomSheetRef,
    title,
    subTitle,
    dismissBtnTitle,
    confirmBtnTitle,
    onDismissBtnHandler,
    onConfirmBtnHandler,
    confirmBtnStyles
}: Props) => {
    return (
        <RBSheet
            ref={bottomSheetRef}
            closeOnDragDown={true}
            animationType={'fade'}
            closeOnPressMask={true}
            customStyles={{
                wrapper: styles.wrapper,
                container: styles.contentContainer,
                draggableIcon: styles.draggableIcon
            }}
        >
            <View style={styles.titlesView}>
                <Text
                    size={Size.Large}
                    fontWeight="bold"
                    color={Colors.text.black_gray}
                    style={styles.title}
                >
                    {title}
                </Text>
                {subTitle && (
                    <Text
                        testID="confirmationDialogueSubTitle"
                        size={Size.XXSmall}
                        color={Colors.text.black}
                        style={styles.subTitle}
                    >
                        {subTitle}
                    </Text>
                )}
            </View>
            <View style={styles.buttonsView}>
                <Button
                    block={true}
                    bordered={false}
                    style={styles.firstButton}
                    onPress={onDismissBtnHandler}
                    testID="dismissBtn"
                >
                    <Text color={Colors.text.primary} fontWeight="600">
                        {dismissBtnTitle}
                    </Text>
                </Button>
                <Button
                    block={true}
                    bordered={false}
                    style={[styles.secondButton, confirmBtnStyles]}
                    onPress={onConfirmBtnHandler}
                    testID="confirmBtn"
                >
                    <Text color={Colors.text.white} fontWeight="600">
                        {confirmBtnTitle}
                    </Text>
                </Button>
            </View>
        </RBSheet>
    );
};

const styles = StyleSheet.create({
    wrapper: {
        backgroundColor: Colors.theme.app_sheet_background_color
    },
    contentContainer: {
        height: 'auto',
        opacity: 1,
        backgroundColor: Colors.extras.white,
        alignItems: 'center',
        borderTopLeftRadius: 40,
        borderTopRightRadius: 40
    },
    draggableIcon: {
        width: 0
    },
    titlesView: {
        width: '80%',
        marginTop: 20
    },
    title: {
        textAlign: 'center'
    },
    subTitle: {
        paddingTop: 20,
        textAlign: 'center'
    },
    buttonsView: {
        width: '100%',
        paddingHorizontal: 20,
        marginTop: 41.86,
        marginBottom: 48
    },
    firstButton: {
        height: 56,
        paddingVertical: 0,
        backgroundColor: Colors.theme.primary_light,
        marginBottom: 16.5
    },
    secondButton: {
        height: 56,
        paddingVertical: 0,
        backgroundColor: Colors.theme.primary,
        borderColor: Colors.button.app_button_primary_border
    }
});

export default ConfirmationDialogue;
