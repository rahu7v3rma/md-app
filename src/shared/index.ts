import Button from './button';
import ChatAvatar from './chatAvatar';
import ChatListItem from './chatListItem';
import ChatManager from './chatManager';
import Checkbox from './checkbox';
import ConfirmationDialogue from './confirmationDialogue';
import CustomCacheImage from './customCacheImage';
import CustomStatusBar from './customStatusBar';
import Header from './header';
import ImageChooser from './imageChooser';
import Input from './input';
import LessonCard from './lessonCard';
import ListItem from './listItem';
import LogBookListItem from './logBookListItem';
import LogInputDropdown from './logInputDropdown';
import LogTab from './logTab';
import LogTimePicker from './logTimePicker';
import LogUnitPicker from './logUnitPicker';
import MainTabBar from './mainTabBar';
import NoDataAvailable from './noDataAvailable';
import PdfViewer from './pdfViewer';
import PlatformImage from './platformImage';
import ProfileImage from './profileImage';
import Progressbar from './progressBar';
import ProgressCircle from './progressCircle';
import Switch from './switch';
import Text from './text';
import VideoPlayer from './videoPlayer';
import Wrapper from './wrapper';

export {
    Button,
    ChatAvatar,
    ChatListItem,
    ChatManager,
    Checkbox,
    ConfirmationDialogue,
    CustomCacheImage,
    CustomStatusBar,
    Header,
    ImageChooser,
    Input,
    LessonCard,
    ListItem,
    LogBookListItem,
    LogInputDropdown,
    LogTab,
    LogTimePicker,
    LogUnitPicker,
    MainTabBar,
    NoDataAvailable,
    PlatformImage,
    ProfileImage,
    Progressbar,
    ProgressCircle,
    Switch,
    Text,
    VideoPlayer,
    Wrapper,
    PdfViewer
};
