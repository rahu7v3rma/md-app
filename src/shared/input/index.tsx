import React, {
    FunctionComponent,
    useCallback,
    useEffect,
    useState
} from 'react';
import {
    KeyboardTypeOptions,
    NativeSyntheticEvent,
    Platform,
    StyleProp,
    StyleSheet,
    TextInput,
    TextInputChangeEventData,
    TextStyle,
    TouchableOpacity,
    View,
    ViewStyle
} from 'react-native';

import Text from '@/shared/text';
import { Colors } from '@/theme/colors';
import { COMMON } from '@/utils/common';
import { Constants } from '@/utils/constants';

import { Size } from '../text';

type Props = {
    placeholder?: string;
    style?: StyleProp<ViewStyle>;
    textInputStyle?: StyleProp<TextStyle>;
    Icon?: React.FunctionComponent<{ color?: string }>;
    secureTextEntry?: boolean;
    required?: boolean;
    errorMessage?: string;
    showError?: boolean;
    eyeColor?: string;
    keyboardType?: KeyboardTypeOptions;
    iconPress?: () => void;
    onChange?: (e: NativeSyntheticEvent<TextInputChangeEventData>) => void;
    onChangeText?: (text: string) => void;
    rightIconPress?: () => void;
    RightIcon?: React.FunctionComponent<{ color?: string }>;
    initialValue?: string;
    editable?: boolean;
    onBlur?: () => void;
    autoCorrect?: boolean;
    autoFocus?: boolean;
    textInputRef?: React.MutableRefObject<TextInput | undefined>;
    multiline?: boolean;
    numberOfLines?: number;
    textInputTestID?: string;
    errorMsgTestID?: string;
    iconPressTestID?: string;
};

const InputComponent: FunctionComponent<Props> = ({
    placeholder = '',
    style,
    textInputStyle,
    Icon,
    secureTextEntry = false,
    required,
    errorMessage = '',
    showError = false,
    keyboardType = 'default',
    iconPress,
    onChange = (_) => {},
    onChangeText = (_) => {},
    rightIconPress,
    RightIcon,
    eyeColor = Colors.icons.eye,
    initialValue = '',
    editable = true,
    onBlur,
    autoCorrect = true,
    autoFocus = false,
    textInputRef,
    multiline = false,
    numberOfLines = 1,
    textInputTestID,
    errorMsgTestID,
    iconPressTestID
}: Props) => {
    const [focused, setFocusState] = useState(false);
    const [inputVal, setInputVal] = useState({
        text: initialValue
    });
    const [showErrorView, setErrorView] = useState(showError);

    useEffect(() => {
        setErrorView(showError);
    }, [showError]);

    const updateInput = useCallback(
        (text: string) => {
            if (text === '') {
                setInputVal({
                    ...inputVal,
                    text
                });
                setErrorView(required ? true : showError);
            } else {
                setInputVal({
                    ...inputVal,
                    text
                });
                setErrorView(showError);
            }
        },
        [inputVal, required, showError]
    );

    const onSubmitEditing = () => {
        if (inputVal.text.trim() === '') {
            setInputVal({
                ...inputVal
            });
            setErrorView(required ? true : showError);
        } else {
            setInputVal({ ...inputVal });
            setErrorView(showError);
        }
    };

    return (
        <View
            style={[
                inputStyle.wrapper,
                showErrorView || showError ? inputStyle.requiredContainer : {},
                showErrorView || showError
                    ? inputStyle.focusedError
                    : focused && inputStyle.focused,
                style
            ]}
        >
            <TouchableOpacity
                activeOpacity={rightIconPress === undefined ? 1 : 0.7}
                onPress={() => {
                    if (rightIconPress === undefined) {
                        return;
                    }

                    rightIconPress();
                }}
            >
                {RightIcon && (
                    <RightIcon
                        color={
                            showErrorView || showError
                                ? Colors.icons.eye_error_border
                                : Colors.icons.eye
                        }
                    />
                )}
            </TouchableOpacity>
            <View style={[inputStyle.container]}>
                <TextInput
                    ref={textInputRef}
                    testID={textInputTestID}
                    placeholder={placeholder}
                    style={[
                        inputStyle.textInput,
                        showErrorView || showError
                            ? inputStyle.textInputError
                            : {},
                        textInputStyle
                    ]}
                    autoFocus={autoFocus}
                    onChange={onChange}
                    placeholderTextColor={
                        showErrorView || showError
                            ? Colors.input.input_placeholder_error
                            : Colors.text.mainDarker
                    }
                    secureTextEntry={secureTextEntry}
                    keyboardType={keyboardType}
                    onChangeText={(text) => {
                        updateInput(text);
                        onChangeText(text);
                    }}
                    onFocus={() => setFocusState(true)}
                    onBlur={() => {
                        if (onBlur) {
                            onBlur();
                        }
                        setFocusState(false);
                        onSubmitEditing();
                    }}
                    onSubmitEditing={() => {
                        onSubmitEditing();
                    }}
                    value={inputVal.text}
                    editable={editable}
                    autoCapitalize="none"
                    autoCorrect={autoCorrect}
                    returnKeyType="done"
                    selectionColor={Colors.text.green}
                    multiline={multiline}
                    numberOfLines={numberOfLines}
                    maxFontSizeMultiplier={Constants.maxFontSizeMultiplier}
                />
                {(showErrorView || showError) && (
                    <Text
                        style={inputStyle.errorText}
                        size={Size.XXXSmall}
                        color={Colors.text.error}
                        fontWeight="500"
                        testID={errorMsgTestID}
                    >
                        {errorMessage}
                    </Text>
                )}
            </View>
            <TouchableOpacity
                testID={iconPressTestID}
                activeOpacity={iconPress === undefined ? 1 : 0.7}
                onPress={() => {
                    if (iconPress === undefined) {
                        return;
                    }

                    iconPress();
                }}
                style={inputStyle.icon}
            >
                {Icon && (
                    <Icon
                        color={
                            showErrorView || showError
                                ? Colors.icons.eye_error_border
                                : !eyeColor
                                ? Colors.text.dark_green
                                : eyeColor
                        }
                    />
                )}
            </TouchableOpacity>
        </View>
    );
};

export default InputComponent;

const inputStyle = StyleSheet.create({
    wrapper: {
        borderWidth: 2,
        borderColor: 'transparent',
        backgroundColor: Colors.extras.white,
        borderRadius: 20,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        paddingHorizontal: 10,
        elevation: Platform.OS === 'android' ? 0 : 2,
        shadowColor: Colors.extras.blackCards,
        shadowOffset: { width: 0, height: 0 },
        shadowRadius: 1,
        shadowOpacity: 0.5,
        height: 61
    },
    container: {
        flexDirection: 'column',
        flex: 1,
        paddingHorizontal: 10,
        position: 'relative',
        justifyContent: 'center'
    },
    requiredContainer: {
        borderColor: Colors.input.error_border,
        backgroundColor: Colors.input.error_bg
    },
    textInput: {
        flex: 1,
        fontFamily: 'Poppins',
        fontWeight: '500',
        color: Colors.text.mainDarker,
        fontSize: 16
    },
    textInputError: {
        paddingBottom: 20
    },
    focused: {
        backgroundColor: Colors.text.white,
        borderColor: Colors.text.green
    },
    focusedError: {
        borderColor: Colors.theme.primary
    },
    errorText: {
        marginBottom: 5,
        position: 'absolute',
        bottom: COMMON.isIos ? 12 : 3,
        left: COMMON.isIos ? 8 : 10,
        zIndex: -1
    },
    icon: {
        marginRight: 5
    }
});
