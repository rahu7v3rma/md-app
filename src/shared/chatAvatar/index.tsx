import React, { FunctionComponent } from 'react';
import { Image, StyleProp, StyleSheet, View, ViewStyle } from 'react-native';

import DefaultAvatar from '@/assets/images/default_avatar.jpg';
import { PlatformImage } from '@/shared';
import { Colors } from '@/theme/colors';

type Props = {
    width?: number;
    height?: number;
    path?: string;
    online?: boolean;
    style?: StyleProp<ViewStyle>;
};

const ChatAvatar: FunctionComponent<Props> = ({
    width,
    height,
    path,
    online,
    style
}: Props) => {
    const borderRadius = width ? width / 2 : 0;

    return (
        <>
            <View style={[styles.wrapper, style]}>
                {path ? (
                    <PlatformImage
                        style={{ borderRadius, width, height }}
                        imageId={path}
                        width={width}
                        height={height}
                    />
                ) : (
                    <Image
                        style={{ borderRadius, width, height }}
                        source={DefaultAvatar}
                    />
                )}
                {online && <View testID="online" style={styles.onlineDot} />}
            </View>
        </>
    );
};

export default ChatAvatar;

const styles = StyleSheet.create({
    wrapper: {
        justifyContent: 'center'
    },
    onlineDot: {
        backgroundColor: Colors.icons.green_check_mark,
        borderStyle: 'solid',
        borderColor: Colors.extras.white,
        width: 12,
        borderWidth: 2,
        height: 12,
        borderRadius: 12,
        zIndex: 1,
        position: 'absolute',
        top: 0,
        transform: [{ translateY: -3 }, { translateX: -3 }]
    }
});
