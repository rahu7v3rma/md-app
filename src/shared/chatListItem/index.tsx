import React, { FunctionComponent } from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';

import { ArrowNext } from '@/assets/svgs';
import { ChatAvatar, Text } from '@/shared';
import { Size } from '@/shared/text';
import { Colors } from '@/theme/colors';
import { LayoutContants } from '@/theme/layoutContants';

type Props = {
    avatar?: any;
    label: string;
    lastMessage?: string;
    time?: string;
    newMessagesCount?: number;
    isOnline?: boolean;
    onPress?: () => void;
    style?: any;
    mainTouchID?: string;
    msgCountID?: string;
};

const ChatListItem: FunctionComponent<Props> = ({
    avatar,
    isOnline = false,
    label,
    lastMessage = '',
    time = '',
    newMessagesCount = 0,
    onPress,
    style,
    mainTouchID,
    msgCountID
}: Props) => {
    return (
        <TouchableOpacity
            testID={mainTouchID}
            onPress={onPress}
            style={[styles.wrapper, style]}
        >
            <View style={styles.firstCol}>
                <View style={styles.avatarView}>
                    <ChatAvatar
                        path={avatar}
                        width={24}
                        height={24}
                        online={isOnline}
                        style={styles.avatar}
                    />
                </View>
            </View>
            <View style={styles.secondCol}>
                <Text
                    size={Size.XSmall}
                    fontWeight="700"
                    color={Colors.text.black}
                >
                    {label}
                </Text>
                {lastMessage !== '' && (
                    <Text size={Size.XXXSmall} color={Colors.text.gray}>
                        {lastMessage.substring(0, 29).concat('...')}
                    </Text>
                )}
            </View>
            <View style={styles.thirdCol}>
                {!time ? (
                    <ArrowNext
                        width={32}
                        height={32}
                        color={Colors.text.purple}
                    />
                ) : (
                    <>
                        <Text size={Size.XXXSmall} color={Colors.text.gray}>
                            {time}
                        </Text>
                        {newMessagesCount > 0 && (
                            <View
                                testID={msgCountID}
                                style={styles.messagesCountView}
                            >
                                <Text color={Colors.text.white}>3</Text>
                            </View>
                        )}
                    </>
                )}
            </View>
        </TouchableOpacity>
    );
};

export default ChatListItem;

const styles = StyleSheet.create({
    wrapper: {
        flexDirection: 'row',
        alignItems: 'center',
        backgroundColor: Colors.extras.white,
        borderRadius: 20,
        borderColor: Colors.button.app_button_disabled_bg,
        height: LayoutContants.cardHeight,
        elevation: 1,
        shadowOpacity: 0.06,
        shadowRadius: 3,
        shadowOffset: { width: 0, height: 0 },
        shadowColor: Colors.theme.app_sheet_background_color
    },
    firstCol: {
        width: '20%',
        alignItems: 'center'
    },
    avatarView: {
        position: 'relative',
        width: 32,
        height: 32,
        borderRadius: 32 / 2
    },
    avatar: {
        width: 32,
        height: 32
    },
    online: {
        position: 'absolute'
    },
    secondCol: {
        width: '60%'
    },
    label: {
        color: Colors.text.black
    },
    thirdCol: {
        width: '20%',
        alignItems: 'center'
    },
    messagesCountView: {
        width: 16,
        height: 16,
        borderRadius: 16 / 2,
        backgroundColor: Colors.theme.primary,
        alignItems: 'center',
        marginTop: 3
    }
});
