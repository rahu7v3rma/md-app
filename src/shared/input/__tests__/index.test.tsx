import React from 'react';
import { StyleSheet } from 'react-native';
import renderer, { act } from 'react-test-renderer';

import { Input } from '@/shared';
import { Colors } from '@/theme/colors';

const inputStyle = StyleSheet.create({
    textInput: {
        flex: 1,
        fontFamily: 'Poppins',
        fontWeight: '500',
        color: Colors.text.mainDarker,
        fontSize: 16
    }
});

it('input snapshot', async () => {
    const tree = renderer.create(<Input />);
    expect(tree).toMatchSnapshot();
});

it('input renders without errors', async () => {
    const tree = renderer.create(<Input />);
    expect(tree.root).toBeTruthy();
});

it('displays placeholder text', async () => {
    const tree = renderer.create(
        <Input textInputTestID="textInput" placeholder={'input placeholder'} />
    );
    let textInput = tree.root.findByProps({
        testID: 'textInput'
    }).props;
    expect(textInput.placeholder).toBe('input placeholder');
});

it('applies custom styling', async () => {
    const tree = renderer.create(
        <Input
            textInputTestID="textInput"
            textInputStyle={{ color: 'white' }}
        />
    );
    let textInput = tree.root.findByProps({
        testID: 'textInput'
    }).props;
    expect(textInput.style).toStrictEqual([
        inputStyle.textInput,
        {},
        { color: 'white' }
    ]);
});

it('handles text input correctly', async () => {
    const tree = renderer.create(<Input textInputTestID="textInput" />);
    let textInput = tree.root.findByProps({
        testID: 'textInput'
    }).props;
    expect(textInput.value).toBe('');
    await act(() => textInput.onChangeText('textValue'));
    let textInputUpdated = tree.root.findByProps({
        testID: 'textInput'
    }).props;
    expect(textInputUpdated.value).toBe('textValue');
});

it('triggers onChange callback', async () => {
    const onChange = jest.fn();
    const tree = renderer.create(
        <Input onChange={onChange} textInputTestID="textInput" />
    );
    let textInput = tree.root.findByProps({
        testID: 'textInput'
    }).props;
    await act(() => textInput.onChange());
    expect(onChange).toHaveBeenCalled();
});

it('displays an error message when showError is true', async () => {
    const tree = renderer.create(
        <Input
            errorMsgTestID={'errorMsg'}
            showError={true}
            errorMessage={'please enter text'}
        />
    );
    let errorText = tree.root.findByProps({
        testID: 'errorMsg'
    });
    expect(errorText).toBeTruthy();
    expect(errorText.props.children).toBe('please enter text');
});

it('displays secure text entry for password input', async () => {
    const tree = renderer.create(
        <Input secureTextEntry={true} textInputTestID="textInput" />
    );
    let textInput = tree.root.findByProps({
        testID: 'textInput'
    }).props;
    expect(textInput.secureTextEntry).toBe(true);
});

it('calls iconPress callback when the icon is pressed', async () => {
    const iconPress = jest.fn();
    const tree = renderer.create(
        <Input
            iconPressTestID={'iconPress'}
            Icon={() => <></>}
            iconPress={iconPress}
        />
    );
    let pressableIcon = tree.root.findByProps({
        testID: 'iconPress'
    }).props;
    await act(() => pressableIcon.onPress());
    expect(iconPress).toHaveBeenCalled();
});
