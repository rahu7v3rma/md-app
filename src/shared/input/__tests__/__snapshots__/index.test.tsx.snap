// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "transparent",
        "borderRadius": 20,
        "borderWidth": 2,
        "elevation": 2,
        "flexDirection": "row",
        "height": 61,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "shadowColor": "rgba(0, 0, 0, 0.3)",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 1,
      },
      {},
      false,
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  />
  <View
    style={
      [
        {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 10,
          "position": "relative",
        },
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={true}
      autoFocus={false}
      editable={true}
      keyboardType="default"
      multiline={false}
      numberOfLines={1}
      onBlur={[Function]}
      onChange={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      placeholder=""
      placeholderTextColor="#4D4F4D"
      returnKeyType="done"
      secureTextEntry={false}
      selectionColor="#72996B"
      style={
        [
          {
            "color": "#4D4F4D",
            "flex": 1,
            "fontFamily": "Poppins",
            "fontSize": 16,
            "fontWeight": "500",
          },
          {},
          undefined,
        ]
      }
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginRight": 5,
        "opacity": 1,
      }
    }
  />
</View>
`;
